src/movies.js

```diff
@@ -1,16 +1,84 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors (movies) {
+    const directors = movies.map(function(movie) {
+        return movie.director
+    });
+    return directors
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies (movies) {
+    const stevenSpielberg = movies.filter(function(movie) {
+        if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')) {
+            return true;
+        }
```

Comment:
Great work Xiaomei! I just want to remind you that filter returns something as long as it is true. Therefore, you don't necessarily need an if statement. Your if statement here could also been written like this: 
return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama').
If anything, i recommend you to check the MDN documentation